src/routes/categories/components/CategoryEntry.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import * as React from 'react'
+
+import { CategoryModel } from '@integreat-app/integreat-api-client'
+import iconPlaceholder from '../assets/IconPlaceholder.svg'
+import styled, { withTheme } from 'styled-components'
+import Highlighter from 'react-highlight-words'
+import normalizeSearchString from '../../../modules/common/utils/normalizeSearchString'
+import Link from 'redux-first-router-link'
+import type { ThemeType } from '../../../modules/theme/constants/theme'
+
+const Row = styled.div`
+  margin: 12px 0;
+
+  & > * {
+    width: 100%;
+  }
+`
+
+const SubCategory = styled.div`
+  text-align: end;
+
+  & > * {
+    width: calc(100% - 60px);
+    text-align: start;
+  }
+`
+
+const CategoryThumbnail = styled.img`
+  width: 40px;
+  height: 40px;
+  flex-shrink: 0;
+  padding: 8px;
+  object-fit: contain;
+`
+
+const CategoryListItem = styled(Highlighter)`
+  display: flex;
+  flex-direction: column;
+  padding: 15px 5px;
+  color: inherit;
+  text-decoration: inherit;
+  height: 100%;
+  min-width: 1px; /* needed to enable line breaks for too long words, exact value doesn't matter */
+  flex-grow: 1;
+  word-wrap: break-word;
+  border-bottom: 2px solid ${props => props.theme.colors.themeColor};
+`
+
+const SubCategoryCaption = styled(CategoryListItem)`
+  margin: 0 15px;
+  padding: 8px 0;
+  border-bottom: 1px solid ${props => props.theme.colors.themeColor};
+`
+const ContentMatchItem = styled(Highlighter)`
+  display: inline-block;
+`
+
+const StyledLink = styled(Link)`
+  display: inline-flex;
+  align-items: center;
+  margin: 0 auto;
+
+  &:hover {
+    color: inherit;
+    text-decoration: inherit;
+    transition: background-color 0.5s ease;
+    background-color: ${props => props.theme.colors.backgroundAccentColor};
+  }
+`
+
+type PropsType = {
+  category: CategoryModel,
+  subCategories: Array<CategoryModel>,
+  /** A search query to highlight in the category title */
+  query?: string,
+  theme: ThemeType
+}
+
+/**
+ * Displays a single CategoryEntry
+ */
+class CategoryEntry extends React.PureComponent<PropsType> {
+  renderSubCategories (): Array<React.Node> {
+    const { subCategories } = this.props
+    return subCategories.map(subCategory =>
+      <SubCategory key={subCategory.hash}>
+        <StyledLink to={subCategory.path}>
+          <SubCategoryCaption searchWords={[]} aria-label={subCategory.title} textToHighlight={subCategory.title} />
+        </StyledLink>
+      </SubCategory>
+    )
+  }
+
+  getMatchedContent (category: CategoryModel): ContentMatchItem {
+    const { query, theme } = this.props
+    if (query && query.length >= 1) {
+      const normalizedFilter = normalizeSearchString(query)
+      const contentWithoutHtml = category.content.replace(/(<([^>]+)>)/ig, '')
+      const matchIdx = contentWithoutHtml.toLowerCase().indexOf(normalizedFilter)
+      if (matchIdx !== -1) {
+        const wordsBeforeMatch = contentWithoutHtml
+          .slice(0, matchIdx)
+          .split(/[\s,]+/)
```

Comment:
Parsing HTML with regex is potentially dangerous. Are you sure this will work always?

Reply:
I also did not like the idea of using a regex here. Found a suitable solution using trim :)

Reply:
Update. Actually you cannot avoid using \s+ here, as we don't want to display e.g. newlines or tabs in the search. And as far as I found out there is like no other suitable solution for that. According to [this package](https://github.com/davisjam/safe-regex) however this regex should be fine. So I would not avoid using it. But feel free to correct me if I am wrong here.